### ISSUE TITLE
 Failed Getting T Magic Value skype4pidgin

### ISSUE THREAD

**Issue by User5959** [EionRobb/skype4pidgin#660]
When i first installed Pidgin i linked my Skype account to it but when i do it, it keeps saying: "Failed getting T Magic value" I keeps getting this tried many other comments and suggestions still no luck. I'm using   Pidgin v 2.14.1 for Windows

**Comment by EionRobb** [comment]
What version of the plugin are you using?  You can find it in, from the buddy list, Help->Plugin Information, then scroll about half-way down the list (it's not alphabetical)

Does it work if you use the "Use alternate auth" setting in the Advanced account settings tab?

Related to https://github.com/EionRobb/skype4pidgin/issues/656 ?

**Comment by jjuanino** [comment]
Hi, I suffer the same issue as User5959 since a few hours (yesterday it connected well) and I confirm that enabling the "Use alternate auth" setting works. I am running 1.7 version. Regards.

**Comment by mkdy** [comment]
Can confirm the issue. Yesterday it was ok.
The alternative method says 'Profile accural is required'.

**Comment by mkdy** [comment]
Can confirm the issue. Yesterday it was ok.
The alternative method says 'Profile accural is required'.

**Comment by boris-petrov** [comment]
Same here, as of today. Getting `Failed getting PPFT value, please try logging in via browser first` - yesterday was working. Using 1.7, compiled a minute ago.

@EionRobb - using the alternative method has been failing all along for me. I get `Login error: Bad username or password (Authentication Failure)`. I've been searching for this for a while now but no one else seems to get that. Any ideas why it might be? I have 2-factor enabled, that's the only thing I imagine can be a problem. Obviously my login (I use my `@live.com` email)/pass are correct because I used to login with them via the "web" method until today.

**Comment by User5959** [comment]
> What version of the plugin are you using? You can find it in, from the buddy list, Help->Plugin Information, then scroll about half-way down the list (it's not alphabetical)
> 
> Does it work if you use the "Use alternate auth" setting in the Advanced account settings tab?
> 
> Related to #656 ?

My Plugin Version is 2.14.1

**Comment by User5959** [comment]
> What version of the plugin are you using? You can find it in, from the buddy list, Help->Plugin Information, then scroll about half-way down the list (it's not alphabetical)
> 
> Does it work if you use the "Use alternate auth" setting in the Advanced account settings tab?
> 
> Related to #656 ?

i don't have "alternative auth" option

**Comment by ZaxonXP** [comment]
I am getting the following message
```
Failed getting PPFT value, please try logging in via browser first
```
Yesterday it was working OK. And I cannot use the `alternative log in  method` as my contacts will not know who is talking to them. Fix trom the #656 was working before but not anymore.

**Comment by bodqhrohro** [comment]
I've done some research; looks like they added one more login stage. Between requests to `/ppsecure/post.srf` and `/login/microsoft`, there should also be a POST request to `https://lw.skype.com/login/oauth/proxy`. `/ppsecure/post.srf` kindly provides a HTML form with parameters to populate it (instead of sFT/opid, that's why parsing them out fails now), so I guess it won't be hard to fix the auth.

**Comment by bodqhrohro** [comment]
Phew, works for me!

It's not actually a *new* stage, but a replacement for the stage introduced after the recent breakage.

I've also made [a DEB package](https://github.com/bodqhrohro/skype4pidgin/releases/download/v1.7.1-alpha/skypeweb-1.7.0-Linux.deb) if someone needs it.

**Comment by ZaxonXP** [comment]
@bodqhrohro: Thanks for the deb package. It is working like a charm. :) +1:

**Comment by bodqhrohro** [comment]
And you know what? That whole stage is now redundant, as the previous one contains the Magic T again! Looks like M$ folks just have rolled back what they messed up in August.

I've just installed the old 1.6.0 build and it logs in fine too.

Check please if that works for you as well. If so, I've just trashed 6 hours of my life for nothing ;D (except of getting more familiar with skypeweb internals, of course).

**Comment by ZaxonXP** [comment]
@bodqhrohro: Indeed, ver. 1.6 works well too. Trough history MS proved that they are able to screw things well. That how they get their bucks trough selling support. :)

**Comment by User5959** [comment]
> Phew, works for me!
> 
> It's not actually a _new_ stage, but a replacement for the stage introduced after the recent breakage.
> 
> I've also made [a DEB package](https://github.com/bodqhrohro/skype4pidgin/releases/download/v1.7.1-alpha/skypeweb-1.7.0-Linux.deb) if someone needs it.

Hello, sorry but can someone make a installer  for Windows?

**Comment by zod6** [comment]
Can confirm that bodqhrohro's patch works.
I wanted to mess with this error myself, but power went off for whole night :D

**Comment by bodqhrohro** [comment]
@User5959 does the [1.6 Windows DLL](https://github.com/EionRobb/skype4pidgin/releases/download/1.6/libskypeweb.dll) work for you? [Here's an instruction](https://developer.pidgin.im/wiki/ThirdPartyPlugins#Windows) on where to put it.

**Comment by jhartotr** [comment]
I'm using skype3pidgin from the master branch (f836eebcda6eba6e321d5baa2efd1934ab43ed1d). Pidgin suddenly fails with

![image](https://user-images.githubusercontent.com/22501936/93687058-081a6f80-fabb-11ea-86a7-760d8aa326e4.png)

I also use the plugin with bitlbee there it also fails:

![image](https://user-images.githubusercontent.com/22501936/93687073-22ece400-fabb-11ea-9ca6-de216411e72f.png)

A few days back it was working fine, should I install version 1.6 of the plugin?

**Comment by User5959** [comment]
> @User5959 does the [1.6 Windows DLL](https://github.com/EionRobb/skype4pidgin/releases/download/1.6/libskypeweb.dll) work for you? [Here's an instruction](https://developer.pidgin.im/wiki/ThirdPartyPlugins#Windows) on where to put it.

Thanks it works but I'm not closing this issue right now to help other people

**Comment by bodqhrohro** [comment]
@jhartotr downgrading to 1.6 fixes the web-based login; you seem to have the alt (SOAP-based) login enabled.

**Comment by jhartotr** [comment]
After switching back to version 1.7:

![image](https://user-images.githubusercontent.com/22501936/93716997-36f41c80-fb73-11ea-9722-cff2ec8b60f7.png)

and disabling "Use alternative login method":

![image](https://user-images.githubusercontent.com/22501936/93716979-0e6c2280-fb73-11ea-8d7b-2024618880bf.png)

it works again:

![image](https://user-images.githubusercontent.com/22501936/93717045-7b7fb800-fb73-11ea-9d5b-9fca5b701fa9.png)

So not sure what happened, I enabled "Use alternative login method" since it was proposed as a solution but didn't work. Probably microsoft was tweaking something on their servers.

**Comment by boris-petrov** [comment]
For me it stopped working *again* (after having reverted to 1.6 a few days ago) - we have to switch to 1.7 yet another time for Skype to login.

**Comment by dlmarquis** [comment]
I also had 1.6 stop working yesterday. 1.7 didn't work for me. What did work, weirdly enough, was adding debug print statements to `skypeweb_string_get_chunk` in skypeweb_util.c immediately before each `g_return_val_if_fail` call. Could there also be a timeout problem somewhere in here?

**Comment by jhartotr** [comment]
After some days working fine with 1.7 it started to fail again for me:

![image](https://user-images.githubusercontent.com/22501936/94318538-0f44f000-ff89-11ea-8418-7f75d9ac57d0.png)

![image](https://user-images.githubusercontent.com/22501936/94318623-38fe1700-ff89-11ea-8300-40f58ac94be0.png)

Login to skype using the browser first didn't help.

Activating "Use alternative login method" fails with a different error.

![image](https://user-images.githubusercontent.com/22501936/94318704-621ea780-ff89-11ea-8bdc-9dfc939a79bb.png)

**Comment by bodqhrohro** [comment]
@jhartotr so do you have now 1.6 working, or both 1.6 and 1.7 don't work?

**Comment by jhartotr** [comment]
I'm using now 1.7. I've now applied the changes on https://github.com/EionRobb/skype4pidgin/pull/665 locally and after recompiling the skypeweb plugin and restarting pidgin it works again.
